src/EditorFeatures/Text/Extensions.cs

```diff
@@ -37,7 +40,7 @@ public static SourceText AsText(this ITextSnapshot textSnapshot)
             => SnapshotSourceText.From(textSnapshot);
 
         internal static SourceText AsRoslynText(this ITextSnapshot textSnapshot, Encoding encoding)
-            => new SnapshotSourceText.ClosedSnapshotSourceText(textSnapshot, encoding);
+            => new SnapshotSourceText.ClosedSnapshotSourceText(((ITextSnapshot2)textSnapshot).TextImage, encoding);
```

Comment:
@heejaechang Are we sure about that? That seems very easy to get broken during a random refactoring.